### DISCUSSION TITLE
Role policy with dynamic arn in the policy

### DISCUSSION THREAD

**Discussion by ghost** [crossplane/crossplane#4058] [General]
Hello, I am trying to add a policy for s3 bucket/kms key or to role and providing static arn works well but i have requirement to add dynamic arn of the resource in the policy. In the example below I can pass the static arn value to the json policy and below is the sample code but i need to provide the kms key arn dynamically that was created using CRD. Please guide me how to do this

apiVersion: iam.aws.upbound.io/v1beta1
kind: Policy
metadata:
  name: demovpc-kms-policy
spec:
  providerConfigRef:
    name: <>
  forProvider:
    tags:
      Name: demovpc-kms-policy
    policy: |
      {
      "Version": "2012-10-17",
      "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "kms:Encrypt",
                "kms:Dycrypt",
                "kms:DescribeKey"
            ],
            "Resource": "arn:aws:kms:us-west-2:<>:key/<>"
        }
      ]
      }

**Comment by bobh66** [comment]
See https://docs.crossplane.io/v1.10/reference/composition/#patching-from-one-composed-rhttps://github.com/upbound/platform-ref-aws/blob/main/package/cluster/network/composition.yamlesource-to-another and also the examples here:
